### PR TITLE
Remove NetCoreReference

### LINKED ISSUE #9168: WPF build projects hand pick the Microsoft.NETCore.App references
Projects in this repository hand pick the targeting pack references that come from the Microsoft.NETCore.App.Ref targeting pack which recently caused issues that required actions:
- https://github.com/dotnet/wpf/commit/d6b1f4560cb72806a9b55beae86ee5f769ec0960
- https://github.com/dotnet/wpf/commit/cb1bc77538a0fe02c4af9a364e515d2f77a5be12
- https://github.com/dotnet/wpf/commit/f704647788c026d99652b5b3716fab92486841d9
- https://github.com/dotnet/wpf/commit/93285ebf1dcd4e40549a001841179d2a1ec286e7

I would like to understand why wpf projects hand pick references. @jaredpar mentioned that there isn't really any benefit in doing. Libraries in runtime that are out-of-band did this as well in the past but we changed that a few years ago.

cc @ericstj

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This removes the `NetCoreReference` infrastructure and replaces it with `DefaultReferenceExclusion` to remove the one problematic implicit reference to WindowsBase.

Manually picking references was blocking using System.Private.Windows.Core from the WinForms repo. Not having this also greatly simplifies the projects.

This also tweaks the solution to add folders docs and eng items.

Fixes #9168
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9914)

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/PresentationUI/PresentationUI.csproj [hunk 1]
```diff
@@ -15,6 +13,7 @@
     <InternalMarkupCompilation>true</InternalMarkupCompilation>
     <NoInternalTypeHelper>true</NoInternalTypeHelper>
     <GenerateDependencyFile>false</GenerateDependencyFile>
+    <DoNotLimitMicrosoftNetCoreReferences>true</DoNotLimitMicrosoftNetCoreReferences>
```
Comment (ThomasGoulet73) [inline src/Microsoft.DotNet.Wpf/src/PresentationUI/PresentationUI.csproj:16]: Why is this needed ? It looks like you removed the handling of this property in the .targets files and this is now the only use `DoNotLimitMicrosoftNetCoreReferences`.
```suggestion
```

File: Microsoft.Dotnet.Wpf.sln [hunk 2]
```diff
@@ -239,21 +239,37 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "OSVersionHelper", "src\Micr
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "PresentationNative", "src\Microsoft.DotNet.Wpf\redist\PresentationNative\PresentationNative.vcxproj", "{AF9084C3-BF37-4A56-A851-89F3BAE731B3}"
 EndProject
-Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{F3F4DFE8-A29C-4BA1-964D-954AB6732744}"
```
Comment (ThomasGoulet73) [inline Microsoft.Dotnet.Wpf.sln:242]: Why are you removing the test projects ?

File: eng/WpfArcadeSdk/tools/SdkReferences.targets [hunk 3]
```diff
@@ -210,4 +125,33 @@
...
+    <DefaultReferenceExclusion Include="WindowsBase"/>
+  </ItemGroup>
+
+  <!--
+    Allows excluding specific .NET runtime libraries from being implicitly referenced.
+    For example, to exclude WindowsBase from being implicitly referenced, add the following to your project file:
+
+      <DefaultReferenceExclusion>
+        <MicrosoftPrivateWinFormsReference Include="WindowsBase" />
+      </DefaultReferenceExclusion>
```
Comment (ThomasGoulet73) [inline eng/WpfArcadeSdk/tools/SdkReferences.targets:140]: This does not look like valid MSBuild syntax. Did you mean something like this ?
```suggestion
      <ItemGroup>
        <DefaultReferenceExclusion Include="WindowsBase" />
      </ItemGroup>
```

File: src/Microsoft.DotNet.Wpf/src/PresentationUI/PresentationUI.csproj (hunk 1 above)
Comment (JeremyKuhne) [reply src/Microsoft.DotNet.Wpf/src/PresentationUI/PresentationUI.csproj:16]: Missed it, thanks :)

File: eng/WpfArcadeSdk/tools/SdkReferences.targets (hunk 3 above)
Comment (JeremyKuhne) [reply eng/WpfArcadeSdk/tools/SdkReferences.targets:140]: Copy past whoops.

File: Microsoft.Dotnet.Wpf.sln (hunk 2 above)
Comment (JeremyKuhne) [reply Microsoft.Dotnet.Wpf.sln:242]: Odd. I wasn't trying to. Must have messed up when I rebased. Fixing.

File: eng/WpfArcadeSdk/tools/SdkReferences.targets [hunk 4]
```diff
@@ -210,4 +125,33 @@
...
+          AfterTargets="ResolveTargetingPackAssets">
+    <ItemGroup>
+      <_referenceExclusion Include="@(DefaultReferenceExclusion)">
+        <AssemblyName>%(DefaultReferenceExclusion.Identity)</AssemblyName>
+        <FrameworkReferenceName>Microsoft.NETCore.App</FrameworkReferenceName>
+        <ExternallyResolved>true</ExternallyResolved>
+      </_referenceExclusion>
+
+      <Reference Remove="@(_referenceExclusion)"
+                 MatchOnMetadata="AssemblyName;FrameworkReferenceName;ExternallyResolved"/>
```
Comment (ericstj) [inline eng/WpfArcadeSdk/tools/SdkReferences.targets:154]: Did you need all of this - I guess you did this so you wouldn't need to intersect `DefaultReferenceExclusion` with `Reference`

Comment (JeremyKuhne) [reply eng/WpfArcadeSdk/tools/SdkReferences.targets:154]: I tried a number of different ways, and this seemed to be the cleanest and easiest to maintain.

File: eng/WpfArcadeSdk/tools/SdkReferences.targets [hunk 5]
```diff
@@ -40,82 +31,6 @@
...
-      <Output ItemName="_netCoreAppReferences" TaskParameter="Result" />
-    </FilterItem1ByItem2>
-
-    <ItemGroup>
-      <Reference Include="@(_netCoreAppReferences->'%(OriginalPath)')" />
-    </ItemGroup>
-  </Target>
-
-  <Target Name="LimitWindowsDesktopAppReferences"
-        AfterTargets="ResolveTargetingPacks;ResolveTargetingPackAsssets"
```
Comment (Kuldeep-MS) [inline eng/WpfArcadeSdk/tools/SdkReferences.targets:83]: why we are removing this target? and how it is connected to .netCoreReference

Comment (JeremyKuhne) [reply eng/WpfArcadeSdk/tools/SdkReferences.targets:83]: It is the same concept for the rest of the SDK. There are no usages. In the testing follow up I have different functionality to never pull from the desktop sdk and use the built binaries only.

**Review by Kuldeep-MS — APPROVED** [review]
